src/lib/net/address.c

```diff
@@ -1693,12 +1693,20 @@ get_interface_address6,(int severity, sa_family_t family, tor_addr_t *addr))
   /* Find the first non-internal address, or the last internal address
    * Ideally, we want the default route, see #12377 for details */
   SMARTLIST_FOREACH_BEGIN(addrs, tor_addr_t *, a) {
+    char *addr_str;
```

Comment:
Style issue: if a variable is only set once, it's often best to put the declaration at the point where it is set, as in `char *addr_str = tor_addr_to_str_dup(addr);`

Same with is_internal, which could be a `const bool`